### ISSUE TITLE
Holding left mouse button breaks selection on mark change

### ISSUE THREAD

**Issue by impulse** [ianstormtaylor/slate#3540]
#### What's the current behavior?
Toggling a mark while holding down the left mouse button will break the selection and force the user to reselect. This issue didn't exist in 0.4x

![Kapture 2020-03-06 at 8 58 54](https://user-images.githubusercontent.com/8146736/76063917-ed523480-5f88-11ea-8001-c62c530319ae.gif)
#### Sandbox:
I can reproduce this in the official example: https://www.slatejs.org/examples/richtext

#### System:
Slate: 0.57.1
Browser: Chrome  / Firefox
OS: Mac

#### What's the expected behavior?
Selection should not break even when the user holds the left mouse button while applying marks

**Comment by CameronAckermanSEL** [comment]
When I try this on chrome on the examples, the selection doesn't break until I try to drag selection left _after_ using a keyboard shortcut to toggle the mark. Is that also what you're seeing?

**Comment by impulse** [comment]
@CameronAckermanSEL yeah I did notice that the issue I showcase in the gif does not happen all that often (in the gif I don't move the mouse at all). The more common problem is that the smallest mouse movement after toggle will break the selection and move the head to the beginning of the word.

![Kapture 2020-03-10 at 8 59 33](https://user-images.githubusercontent.com/8146736/76291490-834dce00-62ad-11ea-90e0-96dd34bbd0bf.gif)

**Comment by codeGun123** [comment]
you can try `ReactEditor.focus(editor);`，  Get the editor to focus again。

**Comment by impulse** [comment]
> you can try `ReactEditor.focus(editor);`， Get the editor to focus again

Just tried, does not work unfortunately
